crowbar_framework/app/constraints/api_constraint.rb

```diff
@@ -15,13 +15,14 @@
 #
 
 class ApiConstraint
-  attr_reader :version
+  attr_reader :major, :minor
 
   def initialize(options)
-    @version = options.fetch(:version)
+    @major = options.fetch(:major)
+    @minor = options.fetch(:minor)
   end
 
   def matches?(request)
-    request.headers.fetch(:accept) == "application/vnd.crowbar.v#{version}+json"
+    request.headers.fetch(:accept) == "application/vnd.crowbar.v#{major}.#{minor}+json"
```

Comment:
`VERSION_MIME = /^application\/vnd\.crowbar\.v(?<major>\d+)\.(?<minor>\d+)\+json$/`

In ruby you can't use variables in constants :-( but with lower case it works.
I have adapted the code a bit with many parts of your proposal, thanks for the hints!